pkg/sql/schema_changer.go

```diff
@@ -1989,42 +2001,39 @@ func (r schemaChangeResumer) Resume(
 		return scErr
 	}
 
-	// For an empty database, the zone config for it was already GC'ed and there's
-	// nothing left to do.
-	if details.DroppedDatabaseID != descpb.InvalidID &&
-		len(details.DroppedTables) == 0 &&
-		len(details.DroppedTypes) == 0 &&
-		len(details.DroppedSchemas) == 0 {
-		return nil
+	// If a database is being dropped and the schema change needs to be completed
+	// in the job, run the schema changer to drop it.
+	if details.FormatVersion >= jobspb.DatabaseJobFormatVersion {
```

Comment:
In order for #52362 to work, this should be done after all of the child objects are done dropping